src/traits/networking/centralized_server_network.rs

```diff
@@ -1064,16 +1070,28 @@ impl<M: NetworkMsg, K: SignatureKey + 'static, E: ElectionConfig + 'static> Conn
 
 /// libp2p identity communication channel
 #[derive(Clone)]
-pub struct CentralizedCommChannel<TYPES: NodeType>(
+pub struct CentralizedCommChannel<
+    TYPES: NodeType,
+    LEAF: LeafType<NodeType = TYPES>,
+    PROPOSAL: ProposalType<NodeType = TYPES>,
+    ELECTION: Election<TYPES>,
```

Comment:
Yes, these generics are unused. I added them because I think we don't want the same Communication Channel to be used for multiple different Proposals/Leaf/Elections. So if we constrain these here, this will make it more difficult to misuse the communication channel. 

An example might help portray what I mean: suppose we've implemented CommunicationChannel for CentralizedCommChannel for both VRF style election and static committees. Then, we accidentally invoke `CommunicationChannel` with VRF style election in one place, then in a separate place we invoke `CommunicationChannel` with static committee election. We meant to use static committee in both places.

Constraining the generics on the struct itself prevents the possibility of that sort of bug, which is why I added it. Not sure if it is worth the extra generics though.

Reply:
Yeah, this makes sense! Let's keep the extra generics then.